### DISCUSSION TITLE
GUI Application for writing Notes

### DISCUSSION THREAD

**Discussion by ap4yk** [community/community#21388] [Other Feature Feedback, Questions, & Ideas]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/9d041bd09cf090b0ef6f68cb6312dd93061dfe35.png" data-download-href="/uploads/short-url/mp1LsZvnEAUFX8l92FxQnDP9a17.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/44062824/181063976-3003a94e-297b-4563-9832-e25eda8b8672.png" alt="image.png" data-base62-sha1="mp1LsZvnEAUFX8l92FxQnDP9a17" width="478" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/9d041bd09cf090b0ef6f68cb6312dd93061dfe35_2_478x500.png, https://aws1.discourse-cdn.com/github/original/2X/9/9d041bd09cf090b0ef6f68cb6312dd93061dfe35.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/9/9d041bd09cf090b0ef6f68cb6312dd93061dfe35.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/9d041bd09cf090b0ef6f68cb6312dd93061dfe35_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">655×685 178 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>Hello fellow programmers. I am writing a GUI application in JAVA that can write notes and read them, add them to a text file and etc.
<p>My problem: GUI Application reads and finds the note.txt file but does not add the writen text to it.</p>
<p>Here is one of the classes from my project that is responsible for adding and reading notes (I think the error comes somewhere from there, shame that I can’t add the whole project here):</p>
<p>import java.io.BufferedWriter;<br>
import java.io.FileWriter;<br>
import java.io.IOException;<br>
import java.util.ArrayList;</p>
<p>/**<br>
*<br>
* <span class="mention">@author</span> justcain<br>
*/<br>
public class AllNotes extends CommonCode {</p>
<p>ArrayList&lt;Note&gt; allNotes = new ArrayList&lt;&gt;();<br>
private String crse = “”;<br>
private int maxID = 0;</p>
<p>AllNotes() {<br>
readAllNotes();<br>
}</p>
<p>public final int getMaxID() {<br>
maxID++;<br>
return maxID;<br>
}</p>
<p>private void readAllNotes() {<br>
ArrayList&lt;String&gt; readNotes = new ArrayList&lt;&gt;();</p>
<p>readNotes = readTextFile(appDir + fileSeparator + “\Notes.txt”);<br>
<a href="//System.out.println" rel="nofollow noopener">//System.out.println</a>(readNotes.get(0));</p>
<p>if (!“File not found”.equals(readNotes.get(0))) {<br>
allNotes.clear();<br>
for (String str : readNotes) {<br>
String<span class="chcklst-box fa fa-square-o fa-fw"></span> tmp = str.split("\t");<br>
System.out.println(str);</p>
<p>Note n = new Note();<br>
n.setNoteID(Integer.parseInt(tmp[0]));<br>
n.setCourse(tmp[1]);<br>
n.setDayte(tmp[2]);<br>
n.setNote(tmp[3]);</p>
<p>allNotes.add(n);</p>
<p>/*int nid = Integer.parseInt(tmp[0]);<br>
Note n = new Note(nid, tmp[1], tmp[2], tmp[3]);<br>
allNotes.add(n);</p>
<p>if (nid &gt; maxID) {<br>
maxID = nid;<br>
}*/</p>
<p>}<br>
}<br>
maxID++;<br>
}</p>
<p>public void addNote(int maxID, String course, String note) {<br>
Note myNote = new Note(maxID, course, note);<br>
allNotes.add(myNote);<br>
writeAllNotes();<br>
}</p>
<p>public ArrayList&lt;Note&gt; getAllNotes() {<br>
return allNotes;<br>
}</p>
<p>private void writeAllNotes() {<br>
String path = appDir + fileSeparator + “\Notes.txt”;<br>
ArrayList&lt;String&gt; writeNote = new ArrayList&lt;&gt;();</p>
<p>for (Note n : allNotes) {<br>
String tmp = n.getNoteID() + “\t”;<br>
tmp += n.getCourse() + “\t”;<br>
tmp += n.getDayte() + “\t”;<br>
tmp += n.getNote();<br>
writeNote.add(tmp);<br>
}<br>
try {<br>
//FileWriter writer = new FileWriter(“path”);<br>
//BufferedWriter bw = new BufferedWriter(writer);<br>
writeTextFile(path, writeNote);<br>
} catch (IOException e) {<br>
System.out.println("Problem! " + path);<br>
}<br>
}</p>
<p>public String searchAllNotesByKeyword(String noteList, int i, String s) {<br>
if (i == allNotes.size()) {<br>
return noteList;<br>
}</p>
<p>if (allNotes.get(i).getNote().contains(s)) {<br>
noteList += allNotes.get(i).getNote() + “\n”;<br>
}</p>
<p>return searchAllNotesByKeyword(noteList, i + 1, s);<br>
}</p>
<p>}</p>

**Comment by Wabri** [comment]
<p>Hi <a class="mention" href="/justcain">@justcain</a>,</p>
<p>Welcome to the Github Community Forum!   </p>
<p>Do you have a github repository? So we can analyze better your code!</p>
<p>-Gabriele-  </p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>

**Comment by ap4yk** [comment]
<p>Just created it. Pretty new to github ;d</p>
<p>Feel free to give advice in any other parts of the code.</p>

**Comment by Wabri** [comment]
<p>Is this the repository? <a href="https://github.com/justcain/GUI-Easy-snippets" rel="nofollow noopener">https://github.com/justcain/GUI-Easy-snippets</a></p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>

**Comment by ap4yk** [comment]
<p>Yes it is. The classes are in the SRC folder.</p>

**Comment by fire-eggs** [comment, marked as answer]
<p>Missing File.CreateNew() in WriteAllText.</p>
